### PR TITLE
feat(Live): add a new resource-live domain name

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

1. add a new resource:  live domain
2. fix mirror code
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/live' TESTARGS='-run=TestAccDomain_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/live -v -run=TestAccDomain_basic -timeout 360m -parallel 4
=== RUN   TestAccDomain_basic
--- PASS: TestAccDomain_basic (1415.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      1415.936s
```

### CODE REVIEW

File: docs/resources/live_domain.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+subcategory: "Live"
+---
+
+# huaweicloud_live_domain
+
+Manages a Live domain within HuaweiCloud.
+
+## Example Usage
+
+### Create a ingest domain name and a streaming domain name
```
Comment (Lance52259) [inline docs/resources/live_domain.md]: s/a/an

File: docs/resources/live_domain.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the Live domain resource. If omitted,
+the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the domain name. Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) Specifies the type of domain name. The options are as follows:
+  - **pull**: streaming domain name.
+  - **push**: ingest domain name.
```
Comment (Lance52259) [inline docs/resources/live_domain.md]: s/-/+

File: docs/resources/live_domain.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `ingest_domain_name` - (Optional, String) Specifies whether to associate an ingest domain name with the streaming
+domain name to push streams to nearby CDN nodes.
+
+* `status` - (Optional, String) Specifies status of the domain name. The options are as follows:
+  - **on**: enable the domain name.
+  - **off**: disable the domain name.
```
Comment (Lance52259) [inline docs/resources/live_domain.md]: ditto.

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+			return fmt.Errorf("error associating the streaming domain name with an ingest domain: %s", err)
+		}
+	}
+
+	return nil
+
+}
+
+func deleteAssociation(d *schema.ResourceData, client *v1.LiveClient) error {
+	pushDomainName, _ := d.GetChange("ingest_domain_name")
```
Comment (Lance52259) [inline huaweicloud/services/live/resource_huaweicloud_live_domain.go]: maybe bring the domain name as an argument is better.

File: docs/resources/live_domain.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/live_domain.md]: done

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:264]: domain name is  more confusing, as the resource is used to create a domain name,

Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go]: done

File: .github/workflows/mirror_to_codehub.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ on:
 
 jobs:
   to_codehub:
-     if: ${{github.repository == 'huaweicloud/terraform-provider-huaweicloud'}}
+    if: ${{github.repository == 'huaweicloud/terraform-provider-huaweicloud'}}
```
Comment (ShiChangkuo) [inline .github/workflows/mirror_to_codehub.yml]: please open another PR to change this

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+	}
+
+	domain := d.Id()
+	response, err := client.ShowDomain(&model.ShowDomainRequest{Domain: &domain})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live domain")
+	}
+
+	r := *response.DomainInfo
+	detail := r[0]
```
Comment (ShiChangkuo) [inline huaweicloud/services/live/resource_huaweicloud_live_domain.go:148]: the index should be checked in an array

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+	domainName := d.Get("name").(string)
+
+	// associate the streaming domain name with an ingest domain Or delete association
+	if d.HasChange("ingest_domain_name") {
+		ingetstDomainNameOld, ingetstDomainName := d.GetChange("ingest_domain_name")
+
+		if ingetstDomainName == "" {
+			err = deleteAssociation(client, domainName, ingetstDomainNameOld.(string))
+		} else {
+			err = associatingDomain(d, client)
```
Comment (ShiChangkuo) [inline huaweicloud/services/live/resource_huaweicloud_live_domain.go:179]: can we associate a new domain without disassociate the old one?

File: docs/resources/live_domain.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the Live domain resource. If omitted,
+the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the domain name. Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/live_domain.md:36]: suggest to add more details about the domain name

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+	d.SetId(createOpts.Body.Domain)
+
+	err = waitingForDomainStatus(ctx, client, d.Id(), model.GetDecoupledLiveDomainInfoStatusEnum().ON,
+		d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// associate the streaming domain name with an ingest domain
+	err = associatingDomain(d, client)
```
Comment (ShiChangkuo) [inline huaweicloud/services/live/resource_huaweicloud_live_domain.go:114]: we can not change the status when creating?

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go (hunk 7 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:179]: yes, of course

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:148]: done

File: docs/resources/live_domain.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+* `type` - (Required, String, ForceNew) Specifies the type of domain name. The options are as follows:
+  + **pull**: streaming domain name.
+  + **push**: ingest domain name.
+
+  Changing this parameter will create a new resource.
+
+* `ingest_domain_name` - (Optional, String) Specifies the ingest domain name, which associates with the streaming
+domain name to push streams to nearby CDN nodes.
+
+* `status` - (Optional, String) Specifies status of the domain name. The options are as follows:
```
Comment (niuzhenguo) [inline docs/resources/live_domain.md:51]: defaults to what

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go (hunk 9 above)
Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:114]: If it is a disabled domain, no need to create it.

File: docs/resources/live_domain.md (hunk 8 above)
Comment (hmbig2) [reply docs/resources/live_domain.md:36]: added

File: huaweicloud/services/live/resource_huaweicloud_live_domain.go (hunk 9 above)
Comment (niuzhenguo) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:114]: if we set status to `off`, you will just ignore it?

Comment (hmbig2) [reply huaweicloud/services/live/resource_huaweicloud_live_domain.go:114]: added

**Pull request merged**